src/main/java/com/github/lookout/metrics/agent/generators/CassandraJMXGenerator.java

```diff
@@ -0,0 +1,121 @@
+        return name.replaceAll("\\W", "_");
+    }
+
+    /**
+     * These are the per-column-family attributes that we will report to statsd
+     */
+    public String[] cfAttributeList = new String[]
+            {   "ReadCount",
+                "WriteCount",
+                "RecentReadLatencyMicros",
+                "RecentWriteLatencyMicros",
+                "TombstonesPerSlice"
+            };
+
+
+    @Override
+    public void generate(StatsDClient statsd) {
+        // Step 1: report PHI values from FailureDetector if available
+        try {
+            TabularData table = (TabularData) mBeanServer.getAttribute(new ObjectName("org.apache.cassandra.net:type=FailureDetector"), "PhiValues");
+            for (Object rawData : table.values() ) {
+                CompositeData data = (CompositeData) rawData;
+                String gaugeName = "phi." + quoteGaugeName(data.get("Endpoint").toString());
+                Double gaugeValue = (Double) data.get("PHI");
+                statsd.gauge(gaugeName, gaugeValue);
+                LOG.debug("Reporting {} as {}", gaugeName, gaugeValue);
+            }
+
+        } catch (Exception e) {
+            LOG.trace("Exception reading PhiValues, skipping", e);
+        }
+
+        // Step 2: per cf statistics
+        Set<ObjectName> names;
+        try {
+            names = mBeanServer.queryNames(new ObjectName("org.apache.cassandra.db:columnfamily=*,keyspace=*,type=ColumnFamilies"), null);
+        } catch (final MalformedObjectNameException e) {
+            throw new RuntimeException(e); // impossible
+        }
+        for (ObjectName name : names) {
+            Long keys = null;
+            String gaugeName = gaugeName(name);
+            try {
+                keys = (Long) mBeanServer.invoke(name, "estimateKeys", new Object[]{}, new String[]{});
+                statsd.gauge(gaugeName + "estimatedKeys", keys);
+            } catch (final InstanceNotFoundException|MBeanException|ReflectionException e) {
+                LOG.debug("Failed to invoke estimateKeys method on {} (ignored)", name, e);
+            }
+            for (String attr : cfAttributeList) {
+                try {
+                    Object value = null;
+                    try {
+                        value = mBeanServer.getAttribute(name, attr);
+                    } catch (final MBeanException|InstanceNotFoundException|ReflectionException e) {
+                        LOG.debug("Couldn't fetch attribute {} from {} (ignored)", new Object[]{attr, name, e});
+                    }
+                    final String fullyQualifiedGaugeName = gaugeName + attr;
+                    if (value instanceof Long) {
+                        statsd.gauge(fullyQualifiedGaugeName, (Long) value);
+                    } else if (value instanceof Double) {
+                        statsd.gauge(fullyQualifiedGaugeName, (Double) value);
+                    } else if (LOG.isWarnEnabled()){
```

Comment:
Perhaps, we only care about Long and Double. But, what if we use "Number" instead of Long & Double.